### PR TITLE
Mine blocks on time travel

### PR DESCRIPTION
**Pull request by njgheorghita**
### What was wrong?
#154 - if I'm not misunderstanding, all that needs to happen is to automine a single block with every `time_travel`. 

### How was it fixed?
Added `mine_blocks` functionality to `time_travel`

#### Cute Animal Picture
image

### CODE REVIEW

File: eth_tester/backends/mock/factory.py [hunk 1]
```diff
@@ -357,7 +357,7 @@ def make_block_from_parent(parent_block, overrides=None):
     if 'timestamp' in overrides:
         yield 'timestamp', overrides['timestamp']
     else:
-        yield 'timestamp', parent_block['timestamp'] + 15
```
Comment (njgheorghita) [inline eth_tester/backends/mock/factory.py]: It seems like this `15` was arbitrary (probably just the avg block time) - as far as I can tell there are no other side affects to changing this.

File: eth_tester/backends/pyevm/main.py [hunk 2]
```diff
@@ -314,7 +314,8 @@ def revert_to_snapshot(self, snapshot):
     # Meta
     #
     def time_travel(self, to_timestamp):
-        self.chain.header = self.chain.header.copy(timestamp=to_timestamp)
+        self.chain.header = self.chain.header.copy(timestamp=(to_timestamp - 1))
```
Comment (njgheorghita) [inline eth_tester/backends/pyevm/main.py]: Timestamps are automatically incremented by 1 sec / mined block [based on](https://github.com/ethereum/py-evm/blob/74a59be41b/eth/vm/base.py#L398)

File: eth_tester/backends/pyevm/main.py [hunk 3]
```diff
@@ -314,7 +314,8 @@ def revert_to_snapshot(self, snapshot):
...
+        self.mine_blocks()
```
Comment (njgheorghita) [inline eth_tester/backends/pyevm/main.py]: The assumption here is that for every `time_travel()` we're only mining one block. Do we want to allow an optional arg to customize how many blocks are mined / time travel?

Comment (carver) [reply eth_tester/backends/pyevm/main.py]: Nah, if they want to mine more, they can just `mine_blocks()` again.

**Pull request merged**